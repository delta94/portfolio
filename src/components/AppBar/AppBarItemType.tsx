import React from 'react';
import { FaCamera, FaCode, FaHome, FaUser } from 'react-icons/fa';
import { IoIosChatbubbles, IoIosSchool } from 'react-icons/io';

export interface AppBarItemType {
  title: string;
  icon: (size?: number, color?: string) => JSX.Element;
}

const appBarItems: AppBarItemType[] = [
  {
    title: 'Home',
    icon: (size?: number, color?: string) => (
      <FaHome size={size ? size : undefined} color={color ? color : undefined} />
    ),
  },
  {
    title: 'About',
    icon: (size?: number, color?: string) => (
      <FaUser size={size ? size : undefined} color={color ? color : undefined} />
    ),
  },
  {
    title: 'Education',
    icon: (size?: number, color?: string) => (
      <IoIosSchool size={size ? size : undefined} color={color ? color : undefined} />
    ),
  },
  {
    title: 'Experience',
    icon: (size?: number, color?: string) => (
      <FaCode size={size ? size : undefined} color={color ? color : undefined} />
    ),
  },
  {
    title: 'Photography',
    icon: (size?: number, color?: string) => (
      <FaCamera size={size ? size : undefined} color={color ? color : undefined} />
    ),
  },
  {
    title: 'Contact',
    icon: (size?: number, color?: string) => (
      <IoIosChatbubbles size={size ? size : undefined} color={color ? color : undefined} />
    ),
  },
];

export { appBarItems };
