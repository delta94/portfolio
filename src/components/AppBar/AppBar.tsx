import clsx from 'clsx';
import React, { useState } from 'react';
import { FiMenu } from 'react-icons/fi';
import { IoMdClose } from 'react-icons/io';
import { createUseStyles } from 'react-jss';
import { useMediaQuery } from 'react-responsive';
import { Link } from 'react-scroll';
import { CustomTheme } from '../../theme';
import { AppBarItem } from '../AppBarItem/AppBarItem';
import { appBarItems, AppBarItemType } from './AppBarItemType';

interface AppBarProps {}

const useStyles = createUseStyles((theme: CustomTheme) => ({
  mainContainer: {
    display: 'flex',
    justifyContent: 'space-evenly',
    position: 'sticky',
    top: 0,
    zIndex: 1,
    backgroundColor: theme.colors.primaryColor,
    paddingTop: theme.spacing(2.5),
    paddingBottom: theme.spacing(2.5),
  },
  mainContainerForMobile: {
    color: theme.colors.white,
    justifyContent: 'flex-end',
    paddingRight: theme.spacing(2.5),
    cursor: 'pointer',
    '&:hover': {
      color: theme.colors.secondaryColor,
      pointerEvent: 'none',
    },
  },
  mobileAppBarItem: {
    marginTop: theme.spacing(1.5),
  },
  mobileAppBarItemsContainer: {
    maxWidth: theme.spacing(32.5),
    width: '100%',
  },
  mobileAppBarItemsMainContainer: {
    display: 'flex',
    justifyContent: 'center',
    marginBottom: theme.spacing(2.5),
  },
  sticky: {
    position: 'sticky',
    top: 0,
    zIndex: 1,
  },
}));

const AppBar: React.FC<AppBarProps> = () => {
  const classes: Record<
    | 'mainContainer'
    | 'mainContainerForMobile'
    | 'mobileAppBarItemsContainer'
    | 'mobileAppBarItem'
    | 'sticky'
    | 'mobileAppBarItemsMainContainer',
    string
  > = useStyles();
  const isMobile: boolean = useMediaQuery({ maxWidth: 780 });
  const [showMobileAppBar, setShowMobileAppBar] = useState<boolean>(true);

  return (
    <div className={classes.sticky}>
      {!isMobile ? (
        <div className={classes.mainContainer}>
          {appBarItems.map((appBarItem: AppBarItemType) => {
            return (
              <Link key={appBarItem.title} to={appBarItem.title} spy={true} smooth={true} offset={50} duration={500}>
                <AppBarItem icon={appBarItem.icon()}>{appBarItem.title}</AppBarItem>
              </Link>
            );
          })}
        </div>
      ) : (
        <>
          {showMobileAppBar ? (
            <div className={clsx([classes.mainContainer, classes.mainContainerForMobile])}>
              <FiMenu onClick={() => setShowMobileAppBar(false)} size={30} />
            </div>
          ) : (
            <>
              <div className={clsx([classes.mainContainer, classes.mainContainerForMobile])}>
                <IoMdClose onClick={() => setShowMobileAppBar(true)} size={30} />
              </div>
              <div className={classes.mobileAppBarItemsMainContainer}>
                <div className={classes.mobileAppBarItemsContainer}>
                  {appBarItems.map((appBarItem: AppBarItemType) => {
                    return (
                      <div className={classes.mobileAppBarItem} key={appBarItem.title}>
                        <AppBarItem fontSize={30} icon={appBarItem.icon(30)}>
                          {appBarItem.title}
                        </AppBarItem>
                      </div>
                    );
                  })}
                </div>
              </div>
            </>
          )}
        </>
      )}
    </div>
  );
};

export { AppBar };
