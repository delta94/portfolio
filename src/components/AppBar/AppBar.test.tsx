import { render } from '@testing-library/react';
import React from 'react';
import { TestContextProvider } from '../../TestContextProvider';
import { AppBar } from './AppBar';

test('AppBar renders correctly', () => {
  const { asFragment } = render(
    <TestContextProvider>
      <AppBar />
    </TestContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
