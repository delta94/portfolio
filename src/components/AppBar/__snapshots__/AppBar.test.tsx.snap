// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="mainContainer-0-2-1"
  >
    <div
      class="mainContainer-0-2-6"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 576 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
        />
      </svg>
      <div
        class="title-0-2-7 title-d0-0-2-8"
      >
        Home
      </div>
    </div>
    <div
      class="mainContainer-0-2-6"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
        />
      </svg>
      <div
        class="title-0-2-7 title-d1-0-2-9"
      >
        About
      </div>
    </div>
    <div
      class="mainContainer-0-2-6"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96.9 270.3V363c0 2.9 1.5 5.5 4 7l132 75.9c5.3 3.1 12-.8 12-7v-93.8c0-2.9-1.5-5.5-4-7l-132-74.9c-5.4-2.9-12 1-12 7.1zM280.9 445.9L413 370c2.5-1.4 4-4.1 4-7v-93.7c0-6.2-6.6-10-12-7l-132 75.9c-2.5 1.4-4 4.1-4 7V439c-.1 6.1 6.6 10 11.9 6.9z"
        />
        <path
          d="M249 65.1L37 188.9c-5.4 3.1-5.4 10.8 0 13.9l212 117.8c4.9 2.8 11 2.8 15.9 0L453 212.9c5.3-3.1 7 .8 7 7v153.4c0 6.8 3.9 10 11 10 4.4 0 10-3.2 10-10V201.5c0-2.9-1.5-5.5-4-7L264.9 65.1c-4.9-2.8-11-2.8-15.9 0z"
        />
      </svg>
      <div
        class="title-0-2-7 title-d2-0-2-10"
      >
        Education
      </div>
    </div>
    <div
      class="mainContainer-0-2-6"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 640 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M278.9 511.5l-61-17.7c-6.4-1.8-10-8.5-8.2-14.9L346.2 8.7c1.8-6.4 8.5-10 14.9-8.2l61 17.7c6.4 1.8 10 8.5 8.2 14.9L293.8 503.3c-1.9 6.4-8.5 10.1-14.9 8.2zm-114-112.2l43.5-46.4c4.6-4.9 4.3-12.7-.8-17.2L117 256l90.6-79.7c5.1-4.5 5.5-12.3.8-17.2l-43.5-46.4c-4.5-4.8-12.1-5.1-17-.5L3.8 247.2c-5.1 4.7-5.1 12.8 0 17.5l144.1 135.1c4.9 4.6 12.5 4.4 17-.5zm327.2.6l144.1-135.1c5.1-4.7 5.1-12.8 0-17.5L492.1 112.1c-4.8-4.5-12.4-4.3-17 .5L431.6 159c-4.6 4.9-4.3 12.7.8 17.2L523 256l-90.6 79.7c-5.1 4.5-5.5 12.3-.8 17.2l43.5 46.4c4.5 4.9 12.1 5.1 17 .6z"
        />
      </svg>
      <div
        class="title-0-2-7 title-d3-0-2-11"
      >
        Experience
      </div>
    </div>
    <div
      class="mainContainer-0-2-6"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M512 144v288c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V144c0-26.5 21.5-48 48-48h88l12.3-32.9c7-18.7 24.9-31.1 44.9-31.1h125.5c20 0 37.9 12.4 44.9 31.1L376 96h88c26.5 0 48 21.5 48 48zM376 288c0-66.2-53.8-120-120-120s-120 53.8-120 120 53.8 120 120 120 120-53.8 120-120zm-32 0c0 48.5-39.5 88-88 88s-88-39.5-88-88 39.5-88 88-88 88 39.5 88 88z"
        />
      </svg>
      <div
        class="title-0-2-7 title-d4-0-2-12"
      >
        Photography
      </div>
    </div>
    <div
      class="mainContainer-0-2-6"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M431 320.6c0-4.4 1.2-8.6 3.3-12.2.6-1.1 1.4-2.1 2.1-3.1 17.4-26 27.6-57.1 27.6-90.3.3-92.2-77.5-167-173.7-167-83.9 0-153.9 57.1-170.3 132.9-2.4 11.1-3.7 22.4-3.7 34.2 0 92.3 74.8 169.1 171 169.1 15.3 0 35.9-4.6 47.2-7.7 11.3-3.1 22.5-7.2 25.4-8.3 2.9-1.1 6.1-1.7 9.3-1.7 3.6 0 7 .7 10.1 2l56.7 20.1s2.4 1 3.9 1c4.4 0 8-3.5 8-8 0-1-.5-2.7-.5-2.7L431 320.6z"
        />
        <path
          d="M318.5 392.5c-3.6 1-8.2 2.1-13.2 3.2-10.5 2.2-23.9 4.5-34 4.5-96.2 0-171-76.8-171-169.1 0-6.6.7-15 1.5-21.4.6-4.3 1.3-8.6 2.3-12.8 1-4.5 2.2-9 3.5-13.4l-8 7.1C66.8 219.2 48 260 48 302.5c0 29.3 8.5 57.5 24.8 82 2.3 3.5 3.6 6.2 3.2 8-.4 1.8-11.9 62-11.9 62-.6 2.9.5 5.8 2.7 7.7 1.5 1.2 3.3 1.8 5.1 1.8 1 0 2-.2 2.9-.6l56.1-22.1c1.8-.7 3.7-1.1 5.7-1.1 0 0 2.4-.2 6.3 1.3 18.9 7.4 39.8 12 60.7 12 46.6 0 90.4-20.1 120.1-55.1 0 0 3.2-4.4 6.9-9.6-3.7 1.3-7.9 2.6-12.1 3.7z"
        />
      </svg>
      <div
        class="title-0-2-7 title-d5-0-2-13"
      >
        Contact
      </div>
    </div>
  </div>
</DocumentFragment>
`;
